plugins/by-name/dap/default.nix

```diff
@@ -62,45 +55,51 @@ with dapHelpers;
     };
   };
 
-  config =
+  # Separate configuration and adapter configurations
+  callSetup = false;
+  extraConfig =
+    cfg:
     let
-      options =
-        with cfg;
-        {
-          inherit configurations;
+      options = {
+        inherit (cfg) configurations;
 
-          adapters =
-            (lib.optionalAttrs (adapters.executables != null) (
-              processAdapters "executable" adapters.executables
-            ))
-            // (lib.optionalAttrs (adapters.servers != null) (processAdapters "server" adapters.servers));
+        adapters =
+          (lib.optionalAttrs (cfg.adapters.executables != null) (
+            dapHelpers.processAdapters "executable" cfg.adapters.executables
+          ))
+          // (lib.optionalAttrs (cfg.adapters.servers != null) (
+            dapHelpers.processAdapters "server" cfg.adapters.servers
+          ));
 
-          signs = with signs; {
-            DapBreakpoint = dapBreakpoint;
-            DapBreakpointCondition = dapBreakpointCondition;
-            DapLogPoint = dapLogPoint;
-            DapStopped = dapStopped;
-            DapBreakpointRejected = dapBreakpointRejected;
-          };
-        }
-        // cfg.extraOptions;
+        signs = with cfg.signs; {
+          DapBreakpoint = dapBreakpoint;
+          DapBreakpointCondition = dapBreakpointCondition;
+          DapLogPoint = dapLogPoint;
+          DapStopped = dapStopped;
+          DapBreakpointRejected = dapBreakpointRejected;
+        };
+      } // cfg.settings;
     in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
-
-      extraConfigLua =
-        (optionalString (cfg.adapters != null) ''
+    {
+      plugins.dap.luaConfig.content = lib.mkMerge [
```

Comment:
Forgot to switch to use the `luaConfig` so lazy loading wasn't putting it in `lz.n` config. :P